### PR TITLE
[Fix] replace np.int with int

### PR DESCRIPTION
**Pull request by jin-s13**
## Motivation

numpy 1.24.0 removes np.bool, np.float, np.int. 
To fix it, I replace np.int (and some np.int64) with np.int32.

## Checklist

**Before PR**:

- [ ] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by jin-s13** [comment]
Same problems exist in xtcocotools, I will fix xtcocotools. 
We may need to upgrade xtcocotools afterwards.

File: mmpose/datasets/datasets/body3d/body3d_mview_direct_panoptic_dataset.py [hunk 1]
```diff
@@ -211,7 +211,7 @@ def _get_db(self):
                             dtype=np.float32)
 
                         cnt = 0
-                        person_ids = -np.ones(self.max_persons, dtype=np.int)
```
Comment (ly015) [inline mmpose/datasets/datasets/body3d/body3d_mview_direct_panoptic_dataset.py:214]: `np.int` was an alias to `int` and would be int64 in a 64-bit system. If so, should we keep this behavior?

Comment (jin-s13) [reply mmpose/datasets/datasets/body3d/body3d_mview_direct_panoptic_dataset.py:214]: ok. let's use int instead.

**Review by ly015 — APPROVED** [review]

**Comment by dancingdarwin** [comment]
Hello, I'd like to kindly follow up on this pull request. I tried downgrading numpy to 1.20.0 but then I get other errors from xtocotools. Thanks so much!

**Pull request merged**